gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsRecord.java

```diff
@@ -0,0 +1,290 @@
+     */
+    NAPTR,
+    /**
+     * Name server record, which delegates a DNS zone to an authoritative server.
+     */
+    NS,
+    /**
+     * Pointer record, which is often used for reverse DNS lookups.
+     */
+    PTR,
+    /**
+     * Start of authority record, which specifies authoritative information about a DNS zone.
+     */
+    SOA,
+    /**
+     * Sender policy framework record, which is used in email validation systems.
+     */
+    SPF,
+    /**
+     * Service locator record, which is used by some voice over IP, instant messaging protocols and
+     * other applications.
+     */
+    SRV,
+    /**
+     * Text record, which can contain arbitrary text and can also be used to define machine readable
+     * data such as security or abuse prevention information.
+     */
+    TXT
+  }
+
+  public static class Builder {
+
+    private List<String> rrdatas = new LinkedList<>();
+    private String name;
+    private Integer ttl;
+    private DnsRecordType type;
+
+    private Builder() {
+    }
+
+    /**
+     * Creates a builder and pre-populates attributes with the values from the provided {@code
+     * DnsRecord} instance.
+     */
+    private Builder(DnsRecord record) {
+      this.name = record.name;
+      this.ttl = record.ttl;
+      this.type = record.type;
+      this.rrdatas.addAll(record.rrdatas);
+    }
+
+    /**
+     * Adds a record to the record set. The records should be as defined in RFC 1035 (section 5) and
+     * RFC 1034 (section 3.6.1). Examples of records are available in Google DNS documentation.
+     *
+     * @see <a href="https://cloud.google.com/dns/what-is-cloud-dns#supported_record_types">Google
+     * DNS documentation </a>.
+     */
+    public Builder addRecord(String record) {
+      this.rrdatas.add(checkNotNull(record));
+      return this;
+    }
+
+    /**
+     * Removes a record from the set. An exact match is required.
+     */
+    public Builder removeRecord(String record) {
+      this.rrdatas.remove(checkNotNull(record));
+      return this;
+    }
+
+    /**
+     * Removes all the records.
+     */
+    public Builder clearRecords() {
+      this.rrdatas.clear();
+      return this;
+    }
+
+    /**
+     * Replaces the current records with the provided list of records.
+     */
+    public Builder records(List<String> records) {
+      this.rrdatas = Lists.newLinkedList(checkNotNull(records));
+      return this;
+    }
+
+    /**
+     * Sets name for this DNS record set. For example, www.example.com.
+     */
+    public Builder name(String name) {
+      this.name = checkNotNull(name);
+      return this;
+    }
+
+    /**
+     * Sets the number of seconds that this record can be cached by resolvers. This number must be
+     * non-negative.
+     *
+     * @param ttl A non-negative number of seconds
+     */
+    public Builder ttl(int ttl) {
+      checkArgument(ttl >= 0, "TTL cannot be negative. The supplied value was " + ttl + ".");
```

Comment:
`checkArgument(ttl >= 0, "TTL cannot be negative. The supplied value was %s.", ttl)`